program-runtime/src/compute_budget.rs

```diff
@@ -198,127 +175,11 @@ impl ComputeBudget {
-                            return Err(duplicate_instruction_error);
-                        }
-                        updated_loaded_accounts_data_size_limit = Some(bytes as usize);
-                    }
-                    _ => return Err(invalid_instruction_data_error),
-                }
-            } else {
-                // only include non-request instructions in default max calc
-                num_non_compute_budget_instructions =
-                    num_non_compute_budget_instructions.saturating_add(1);
-            }
-        }
-
-        if let Some((bytes, i)) = requested_heap_size {
-            if bytes > MAX_HEAP_FRAME_BYTES
-                || bytes < MIN_HEAP_FRAME_BYTES as u32
-                || bytes % 1024 != 0
-            {
-                return Err(TransactionError::InstructionError(
-                    i,
-                    InstructionError::InvalidInstructionData,
-                ));
-            }
-            self.heap_size = bytes;
-        }
-
-        let compute_unit_limit = updated_compute_unit_limit
-            .unwrap_or_else(|| {
-                num_non_compute_budget_instructions
-                    .saturating_mul(DEFAULT_INSTRUCTION_COMPUTE_UNIT_LIMIT)
-            })
-            .min(MAX_COMPUTE_UNIT_LIMIT);
-        self.compute_unit_limit = u64::from(compute_unit_limit);
-
-        self.loaded_accounts_data_size_limit = updated_loaded_accounts_data_size_limit
-            .unwrap_or(MAX_LOADED_ACCOUNTS_DATA_SIZE_BYTES)
-            .min(MAX_LOADED_ACCOUNTS_DATA_SIZE_BYTES);
-
-        Ok(prioritization_fee
-            .map(|fee_type| PrioritizationFeeDetails::new(fee_type, self.compute_unit_limit))
-            .unwrap_or_default())
-    }
-
-    pub fn fee_budget_limits<'a>(
-        instructions: impl Iterator<Item = (&'a Pubkey, &'a CompiledInstruction)>,
-        feature_set: &FeatureSet,
-    ) -> FeeBudgetLimits {
-        let mut compute_budget = Self::default();
-
-        let prioritization_fee_details = compute_budget
-            .process_instructions(
-                instructions,
-                !feature_set.is_active(&remove_deprecated_request_unit_ix::id()),
-                feature_set.is_active(&add_set_tx_loaded_accounts_data_size_instruction::id()),
-            )
-            .unwrap_or_default();
-
-        FeeBudgetLimits {
-            loaded_accounts_data_size_limit: compute_budget.loaded_accounts_data_size_limit,
-            heap_cost: compute_budget.heap_cost,
-            compute_unit_limit: compute_budget.compute_unit_limit,
-            prioritization_fee: prioritization_fee_details.get_fee(),
-        }
+    pub fn new_from_compute_budget_limits(compute_budget_limits: &ComputeBudgetLimits) -> Self {
+        let mut compute_budget =
+            ComputeBudget::new(u64::from(compute_budget_limits.compute_unit_limit));
+        compute_budget.heap_size = compute_budget_limits.updated_heap_bytes;
+        compute_budget
     }
```

Comment:
instead of having to process instructions to furnish ComputeBudget, it can be new-ed from ComputeBudgetLimits processed (only once) and stored elsewhere.